CONTRIBUTING.rst

```diff
@@ -0,0 +1,177 @@
+
+.. code-block:: sh
+
+   git add <file_I_created> <and_file_I_modified>
+
+Commit & push your changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Periodically commit your changes with
+
+.. code-block:: sh
+
+   git commit -m "This is a 50-char description to explain my work"
+
+The commit message (between quotation marks) is super important in order to follow the developments and identify bugs.
+
+For the moment, commits are on your local repo only.
+You can push them to your fork with
+
+.. code-block:: sh
+
+   git push -u origin <branch_name>
+
+If you want to synchronize your branch with the ``master`` branch (this is useful when the ``master`` branch is being modified while you are working on ``<branch_name>``), you can use
+
+.. code-block:: sh
+
+   git pull upstream master
+
+and fix any conflict that may occur.
+
+Submit a Pull Request
+^^^^^^^^^^^^^^^^^^^^^
+
+A Pull Request (PR) is the way to efficiently visualize the changes you made and to propose your new feature/improvement/fix to the WarpX project.
+Right after you push changes, a banner should appear on the Github page of your fork, with your ``<branch_name>``.
+
+- Click on the ``compare & pull request`` button to prepare your PR.
+- It is time to communicate your changes: write a title and a description for your PR.
+  People who review your PR are happy to know
+
+  * what feature/fix you propose, and why
+  * how you made it (created a new class than inherits from...)
+  * and anything relevant to your PR (performance tests, images, *etc.*)
+- Press ``Create pull request``.
+  Now you can navigate through your PR, which highlights the changes you made.
+
+Please DO NOT write large Pull Requests, as they are very difficult and time-consuming to review.
+As much as possible, split them into small targeted PRs.
+For example, if find typos in the documentation open a pull request that only fixes typos.
+If you want to fix a bug, make a small pull request that only fixes a bug.
+If you want to implement a large feature, write helper functionality first, test it and submit those as a first pull request.
+If you want to implement a feature and are not too sure how to split it, just open an issue about your plans and ping other WarpX developers on it to chime in.
+
+Even before your work is ready to merge, it can be convenient to create a PR (so you can use Github tools to visualize your changes).
+In this case, please put the ``[WIP]`` tag (for Work-In-Progress) at the beginning of the PR title.
+
+Include a test to your PR
+"""""""""""""""""""""""""
+
+A new feature is great, a **working** new feature is even better!
+Please test your code and add your test to the automated test suite.
+It's the way to protect your work from adventurous developers.
+Instructions are given in the :ref:`testing section <developers-testing>` of our `developer's documentation <https://warpx.readthedocs.io/en/latest/developers/testing.html>`_.
+
+Include documentation to your PR
```

Comment:
Sorry, I am really not sure why or how to help, but it reads a little strangely. Maybe:
```suggestion
Include documentation about your PR
```